react-native/react/login/register/code-page/index.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+/*
+ * Screen to scan/show qrcode/text code. Goes into various modes with various options depending on if
+ * you're a phone/computer and if you're the existing device or the new device
+ */
+
+import React, {Component} from '../../../base-react'
+import {connect} from '../../../base-redux'
+import {codePageDeviceRoleExistingPhone, codePageDeviceRoleNewPhone,
+        codePageDeviceRoleExistingComputer, codePageDeviceRoleNewComputer} from '../../../constants/login'
+import Render from './index.render'
+
+class CodePage extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      enterText: ''
+    }
+  }
+
+  render () {
+    return (
+      <Render
+        onChangeText={enterText => this.setState({enterText})}
```

Comment:
This is a pattern we should adopt, we should hoist up this kind of logic into the smart component and not leave it in both renderers